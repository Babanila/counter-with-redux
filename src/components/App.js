import React from "react"
import { useSelector, useDispatch } from "react-redux"
import { increment, decrement, cancel } from "../actions"
import "../styles/App.css"

function App() {
  const counter = useSelector(state => state.counter)
  const dispatch = useDispatch()
  return (
    <div>
      <h1>Simple Counter App</h1>
      <h3>Counter {counter}</h3>
      <button onClick={() => dispatch(increment())}>+</button>
      <button onClick={() => dispatch(decrement())}>-</button>
      <button id="cancelButton" onClick={() => dispatch(cancel())}>
        reset
      </button>
    </div>
  )
}

export default App
