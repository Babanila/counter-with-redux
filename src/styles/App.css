button {
  width: 3rem;
  background-color: #1fd82f;
  color: #ffffff;
  align-self: center;
  font-size: 1rem;
  padding: 10px;
  margin-left: 1rem;
}

button:hover {
  cursor: pointer;
  background-color: #ffffff;
  color: #000000;
}

#cancelButton {
  width: 6rem;
  background-color: #ff0000;
  color: #ffffff;
  font-size: 1rem;
  align-self: center;
  padding: 10px;
  margin-left: 1rem;
}

#cancelButton:hover {
  cursor: pointer;
  background-color: #ffffff;
  color: #000000;
}
