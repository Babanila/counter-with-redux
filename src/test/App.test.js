import React from "react"
import { shallow } from "enzyme"
import App from "../components/App"

describe("App Test", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it("should render and match the snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error")
    expect(wrapper).not.toBeNull()
    expect(spy).not.toHaveBeenCalled()
  })
})
