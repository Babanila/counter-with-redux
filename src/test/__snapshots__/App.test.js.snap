// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test should render and match the snapshot 1`] = `
<div>
  <h1>
    Simple Counter App
  </h1>
  <h3>
    Count: 
    0
  </h3>
  <ButtonGroup />
</div>
`;
